tests/deploy/ui/test_ui.py

```diff
@@ -0,0 +1,183 @@
+import os
+import time
+from multiprocessing import Process
+
+import httpx
+import panel as pn
+import pytest
+import uvicorn
+from playwright.sync_api import expect, sync_playwright
+
+from ragna.assistants import RagnaDemoAssistant
+from ragna.core._utils import default_user
+from ragna.deploy import ApiConfig, Config, UiConfig
+from ragna.deploy._api import app as api_app
+from ragna.deploy._ui import app as ui_app
+
+TEST_API_PORT = "38769"
+TEST_UI_PORT = "38770"
```

Comment:
Instead of hardcoding them, can we just make them fully random? See 

https://github.com/Quansight/ragna/blob/dfb434812ad4931f54181e69e7cf04c99351d629/tests/utils.py#L79-L82

(note that this is an old commit and we removed this in #322)